src/org/traccar/database/DeviceManager.java

```diff
@@ -387,4 +402,54 @@ public void sendCommand(Command command) throws Exception {
         }
         return result;
     }
+
+    public void updateMotionState(long deviceId, Position position) {
```

Comment:
We can't `MotionHandler` are multiple for every pipeline, but we need some singleton to store per device motion states. Or am I mistaken?